Reed/Reader/views/ReaderView.swift

```diff
@@ -42,12 +53,22 @@ struct ReaderView: View {
                     width: UIScreen.main.bounds.width,
                     height: UIScreen.main.bounds.height * 0.65
                 ))
-                Text("\(viewModel.curPage + 1) of \(viewModel.pages.endIndex)")
+                if viewModel.section?.prevNcode == nil && viewModel.section?.nextNcode == nil {
+                    Text(viewModel.curPage == -1 ? "" : "\(viewModel.curPage + 1) of \(viewModel.pages.endIndex)")
+                } else if viewModel.section?.prevNcode == nil {
+                    Text(viewModel.curPage == -1 || viewModel.curPage + 1 > viewModel.pages.endIndex - 1 ? "" : "\(viewModel.curPage + 1) of \(viewModel.pages.endIndex - 1)")
+                } else if viewModel.section?.nextNcode == nil {
+                    Text(viewModel.curPage <= 0 ? "" : "\(viewModel.curPage) of \(viewModel.pages.endIndex - 1)")
+                } else {
+                    Text(viewModel.curPage <= 0 || viewModel.curPage > viewModel.pages.endIndex - 2 ? "" : "\(viewModel.curPage) of \(viewModel.pages.endIndex - 2)")
+                }
```

Comment:
Abstract this into a function in the view model that calculates the correct Text value.

Reply:
fixed